PiwikTracker/Queue.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  EventQueue.swift
+//  PiwikTracker
+//
+//  Created by Cornelius Horstmann on 09.10.16.
+//  Copyright © 2016 Mattias Levin. All rights reserved.
+//
+
+import Foundation
+
+/// An Element that is queueable.
+protocol Queueable {
+    /// A unique identifier. It should be unique per object.
+    var id: UUID { get }
+    
+    /// This property is used to get the data that should be stored in the queue.
+    var data: Data { get }
+    
+    /// This initializer is used when dequeueing.
+    init?(data: Data)
```

Comment:
Right now the Queue can hold only one type of objects. So this would have worked. I thought that is sufficient to store all events, but now that I am writing this I see that it cripples the queue and limits its usage. Say we want to have different types of events (download event, view event, ...). We might decide to have different objects for those. I will refactor the queue to store all multiple types of objects.